src/silo/config/config_repository.test.cpp

```diff
@@ -30,33 +30,37 @@ ConfigReaderMock mockConfigReader(const DatabaseConfig& config) {
 
 TEST(ConfigRepository, shouldReadConfigWithoutErrors) {
    const auto config_reader_mock = mockConfigReader(
-      {"main",
-       {
-          "testInstanceName",
+      {.default_nucleotide_sequence = "main",
+       .schema =
           {
-             {"testPrimaryKey", ValueType::STRING},
-             {"metadata1", ValueType::PANGOLINEAGE, true},
-             {"metadata2", ValueType::DATE},
-          },
-          "testPrimaryKey",
-          std::nullopt,
-          "metadata1",
-       }}
+             .instance_name = "testInstanceName",
+             .metadata =
+                {
+                   {.name = "testPrimaryKey", .type = ValueType::STRING},
+                   {.name = "metadata1", .type = ValueType::PANGOLINEAGE, .generate_index = true},
+                   {.name = "metadata2", .type = ValueType::DATE},
+                },
+             .primary_key = "testPrimaryKey",
+             .date_to_sort_by = std::nullopt,
+             .partition_by = "metadata1",
+          }}
    );
 
    ASSERT_NO_THROW(ConfigRepository(config_reader_mock).getValidatedConfig("test.yaml"));
 }
 
 TEST(ConfigRepository, shouldThrowIfPrimaryKeyIsNotInMetadata) {
    const auto config_reader_mock = mockConfigReader(
-      {"main",
-       {
-          "testInstanceName",
+      {.default_nucleotide_sequence = "main",
+       .schema =
           {
-             {"notPrimaryKey", ValueType::STRING},
-          },
-          "testPrimaryKey",
-       }}
+             .instance_name = "testInstanceName",
+             .metadata =
+                {
+                   {.name = "notPrimaryKey", .type = ValueType::STRING},
+                },
+             .primary_key = "testPrimaryKey",
+          }}
```

Comment:
Oh nice, I didn't know that C++ has named arguments. This looks way better than before.

Reply:
A newer cpp20 feature: https://pdimov.github.io/blog/2020/09/07/named-parameters-in-c20/